### PR TITLE
V0.1.0

### LINKED ISSUE #7: numpy 2.0 compatibility
related to tataratat/splinepy#440

needs newer version of pybind11

### PR DESCRIPTION
**Pull request by j042**
- build updates/modernization
- pybind11 update -> compatible with numpy2.0
- add `__version__`
- fixed #7

### CODE REVIEW

**Comment by j042** [comment]
@clemens-fricke we can add funi's version to splinepy cli after this

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+project(
+  funi
+  VERSION 0.1.0
+  LANGUAGES CXX)
 
 if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+
 endif()
 
 # options - default is header only
```
Comment (clemens-fricke) [inline CMakeLists.txt:11]: Is this comment correct? I see this as the default value for this option is 'ON', meaning ¿true?.

**Review by clemens-fricke — APPROVED** [review]
It looks good to me. Check if the comment I highlighted is correct; otherwise, it seems okay. Thank you.

File: CMakeLists.txt (hunk 1 above)
Comment (j042) [reply CMakeLists.txt]: good catch, updated!

**Pull request merged**